internal/clients/team4/judge.go

```diff
@@ -33,16 +35,66 @@ type judge struct {
 
 // InspectHistory is the base implementation of evaluating islands choices the last turn.
 // OPTIONAL: override if you want to evaluate the history log differently.
+type valuePair struct {
+	actual   shared.Resources
+	expected shared.Resources
+}
+
+type judgeHistoryInfo struct {
+	Resources  valuePair // amount of resources available and reported by the island
+	Taxation   valuePair // amount of tax paid and expected
+	Allocation valuePair // amount of allocation taken and granted
+	Lied       int       // number of times the island has lied
+}
+
+func (j *judge) saveHistoryInfo(iigoHistory *[]shared.Accountability, lieCounts *map[shared.ClientID]int, turn uint) {
+	accountabilityMap := map[shared.ClientID][]rules.VariableValuePair{}
+	for _, acc := range *iigoHistory {
+		client := acc.ClientID
+		if accountabilityMap[client] != nil {
```

Comment:
Any chance there is a cleaner way of doing this? Can we initialise the list with all clientID's maybe?

Reply:
done